### ISSUE TITLE
Migrate from Kotlin synthetics to Jetpack view binding

### ISSUE THREAD

**Issue by ialokim** [grote/Transportr#737]
See https://developer.android.com/topic/libraries/view-binding/migration and https://developer.android.com/topic/libraries/view-binding

**Comment by AcharyaS97** [comment]
I've started on this!

**Comment by AcharyaS97** [comment]
So I had a question about this. I want to finish this up but having issues with some of the UI pieces (PickTransportNetworkActivity, AboutActivity, AboutFragment, ContributorFragment) that rely on external dependencies for some ui binding. The ui classes in question use some of the mikepenz dependencies:


	implementation 'com.mikepenz:aboutlibraries:6.2.0'
	implementation "com.mikepenz:fastadapter:$fastadapterVersion"
	implementation "com.mikepenz:fastadapter-commons:$fastadapterVersion"
	implementation "com.mikepenz:fastadapter-extensions-expandable:5.3.4@aar"
	
I want to update their versions to the latest because they have been updated to support viewbinding, but I'm a little confused how to do that given the fact the project uses the gradle witness plugin. Would I need to manually update the witness file after updating the dependencies so that the checksum on each updated dependency matches what I see on the versions I update to?

**Comment by grote** [comment]
Run this script after bumping the dependencies: https://github.com/grote/Transportr/blob/master/update-dependency-pinning.sh

**Comment by ialokim** [comment]
Be aware that Mike Penz' libraries tend to introduce API breaking changes with new versions. So be sure to test thoroughly all parts of the app using these libraries...

**Comment by AcharyaS97** [comment]
@grote Great thanks i'll do that.

@ialokim So then should I add automated tests for related pieces as a part of that?

**Comment by AcharyaS97** [comment]
Also, where are all the variables defined that are used as placeholders for dependency versions?

**Comment by ialokim** [comment]
> Also, where are all the variables defined that are used as placeholders for dependency versions?

Did you mean [these](https://github.com/grote/Transportr/blob/master/build.gradle#L4)?

**Comment by AcharyaS97** [comment]
yeah. sorry i haven't forgotten about this. i've been having issues with some of the mikepenz libraries and updating those to use viewbinding.

**Comment by ialokim** [comment]
No problem. Would you want to continue working on it or could you push the changes you already did somewhere so someone else could take over?

**Comment by AcharyaS97** [comment]
I wanted to continue working on it. When did you want me to have it in by?

**Comment by ialokim** [comment]
We have no deadline. I just wanted to make sure that the work that you had already done would not get lost.

I imagine that updating the mikepenz libraries is a big task. Not sure whether we could do the migration in parts, i.e., leaving the fastadapter library updates apart for now?
